packages/labs/router/README.md

```diff
@@ -2,6 +2,255 @@
 
 A router for Lit.
 
+## Status
+
+`@lit-labs/router` is not yet published to npm.
+
+This package requires either a native [`URLPattern`](https://developer.mozilla.org/en-US/docs/Web/API/URLPattern) implementation (which is currently only implemented in Chrome, Edge, and other Chromium browsers) or a URLPattern polyfill, like [`urlpattern-polyfill`](https://github.com/kenchris/urlpattern-polyfill).
+
 ## Overview
 
-_This project is just starting. Details TBD._
+`@lit-labs/router` is a component-oriented router API vended as reactive controllers. Routes are configured as part of component definitions, and integrated into the component lifecycle and rendering.
+
+Usage will generally look like this, with a configuration in a reactive controller, and rendering done via route-specific render callbacks and an "outlet" to use in the main render() method:
+
+```ts
+class MyElement extends LitElement {
+  private _routes = new Routes(this, [
+    {path: '/', render: () => html`<h1>Home</h1>`},
+    {path: '/projects', render: () => html`<h1>Projects</h1>`},
+    {path: '/about', render: () => html`<h1>About</h3>`},
+  ]);
+
+  render() {
+    return html`
+      <header>...</header>
+      <main>${this._routes.outlet}</main>
+      <footer>...</footer>
+    `;
+  }
+}
+```
+
+Routes can be nested: a route path can include a trailing `/*` pattern to match against a prefix, and will automatically propagate that prefix to Routes controllers defined in child elements.
+
+The general shape of the API includes:
+
+- A `Router` controller that's used as a top-level singleton to set up event listeners
+- A `Routes` controller for declaring routes inside components
+- Declaration of routes with [`URLPattern`](https://developer.mozilla.org/en-US/docs/Web/API/URLPattern) and render callbacks
+- Extraction of URL pattern parameters into data objects passed to render callbacks
+- A `routes.outlet` that renders the current route's render callback
+- A `routes.link()` method to generate URLs to use in `<a>` tags, etc.
+- A `routes.goto()` method for performing a navigation
+
+## API
+
+### Router
+
+A Router is a controller (a subclass of the Routes controller) for use at the top-level of an application. It's main purpose is to set up global `click` and `popstate` event listeners (which should be installed only once on a page). It can optionally contain route definitions as a convenience.
```

Comment:
I had considered making `Router` _not_ extend `Routes` and just be a thing that installs the event listeners. For simple cases though a component would need to install both a `Router` and a `Routes`, and that seemed unfortunate.

React Router also has this Router/Routes division. In their case they have browser, server, and browser/fragment implementations. We're going to try to run the same router on the client and server, but one could imagine a Router implementation that uses the App History API instead of catching click and popstate events.

I'm hoping we can document that "Router goes with the app, Routes go with nested components" and that'll make sense.

Reply:
Sounds good.